arch/x86/core/Kconfig.ia32

```diff
@@ -137,8 +137,8 @@ endif # CPU_HAS_FPU
 
 config X86_FP_USE_SOFT_FLOAT
 	bool
-	prompt "Use Software Floating Point Operations" if !(NEWLIB_LIBC && !FPU)
-	default y if NEWLIB_LIBC && !FPU
+	prompt "Use Software Floating Point Operations" if !FPU
+	default y if !FPU
```

Comment:
@dcpleung since you introduced this requirement initially, does this change sound reasonable?

FYI, all the other archs are using soft float when `CONFIG_FPU=n`.

Reply:
I don't understand why `CONFIG_X86_USE_SOFT_FLOAT` is even configurable by the user here; if you are using the FPU, Zephyr needs to know this, so `CONFIG_FPU` must be the inverse of `CONFIG_X86_USE_SOFT_FLOAT`.

Reply:
This looks fine.

IIRC, there was some size differences if soft float is enabled even though floating point is never used. This is due to different GCC libraries being included. We might want to remove the prompt so that this behaves the same as other arch.

Reply:
Or maybe this comes back having a Kconfig that does the right thing

e.g.

CONFIG_FLOAT with options

NO_FLOAT (no fpu usage, no soft float flags)
SOFT_FLOAT (no fpu usage, explicit soft float compiler flags)
HARD_FLOAT (fpu usage, error if HAS_FPU=n)
AUTO_FLOAT (use FPU if available, or soft float flags if not)

If the size output is actually different when not using floats and linking in soft float then that seems like we should have something like this potentially? Or if this isn't useful then we should do the above and note in the docs CONFIG_FPU=n implies soft float inclusion if the toolchain supports it (presumably does).

Reply:
How would `NO_FLOAT` differ from `SOFT_FLOAT`? Do you expect the build system to generate errors if the application accidentally uses a bit of floating point math?

How would `HARD_FLOAT` differ from `AUTO_FLOAT`? Would you expect the system to fail to build if there wasn't an FPU?

I think the existing option to force soft float is exactly right; when set, you never use the FPU. When not set, you use any available FPU. Whether it's soft float by default is a separate question; I generally don't buy SoCs with an FPU if I'm not going to use it, but I don't know what other people do.

Reply:
NO_FLOAT comes from my assumption the current x86 setup is there for a reason which is effectively the default today. The compiler will use the FPU instructions but result in an exception. The reasoning is code size being smaller from the explanation above by @dcpleung.

HARD_FLOAT would fail if CPU_HAS_FPU=n. 

AUTO_FLOAT should use the FPU if it exists, if not use software floating point from the toolchain with whatever flags that entails.

Reply:
I just removed the prompt of CONFIG_X86_USE_SOFT_FLOAT to make it non-configurable by user.